clusterloader2/pkg/framework/client/objects.go

```diff
@@ -172,7 +180,14 @@ func CreateObject(dynamicClient dynamic.Interface, namespace string, name string
 		_, err := dynamicClient.Resource(gvr).Namespace(namespace).Create(obj, metav1.CreateOptions{})
 		return err
 	}
-	return RetryWithExponentialBackOff(RetryFunction(createFunc, apierrs.IsAlreadyExists))
+	return RetryWithExponentialBackOff(createRetryFunction(
+		createFunc,
+		apierrs.IsAlreadyExists /* isAllowedError */,
+		// We retry the 404 errors for the create object operations as they may be caused by races,
+		// e.g. when a ServiceAccount is created the associated token is created asynchronously, when
+		// creating an object that needs the token we may get retryable 404 error if the token has not
+		// been created yet.
+		logError(apierrs.IsNotFound) /* isRetryableError */))
```

Comment:
I don't agree with that fix - while it may make sense in some context, it's not correct in generic case.

If we really need to tolerate some other errors in some contexts, we should make the interface to also accept allowed errors (and default to just Already Exists or sth like that).